libs/unity/library/Tests/Runtime/MediaLineTests.cs

```diff
@@ -235,5 +233,66 @@ public IEnumerator SetReceiver()
             // Terminate the coroutine.
             yield return null;
         }
+
+        [UnityTest]
+        public IEnumerator DestroyPeerConnection()
+        {
+            // Create the component
+            var go = new GameObject("test_go");
+            go.SetActive(false); // prevent auto-activation of components
+            var pc = go.AddComponent<PeerConnection>();
+
+            // Add a media line while inactive.
+            VideoTrackSource source1 = go.AddComponent<UniformColorVideoSource>();
+            VideoReceiver receiver1 = go.AddComponent<VideoReceiver>();
+            MediaLine ml1 = pc.AddMediaLine(MediaKind.Video);
+            ml1.Source = source1;
+            ml1.Receiver = receiver1;
+
+            // Media lines have not been set yet.
+            Assert.IsEmpty(source1.MediaLines);
+            Assert.IsNull(receiver1.MediaLine);
+
+            yield return PeerConnectionTests.InitializeAndWait(pc);
+
+            // Media lines have been set now.
+            Assert.AreEqual(source1.MediaLines.Single(), ml1);
+            Assert.AreEqual(receiver1.MediaLine, ml1);
+
+            // Add a media line while active.
+            VideoReceiver receiver2 = go.AddComponent<VideoReceiver>();
+            MediaLine ml2 = pc.AddMediaLine(MediaKind.Video);
+            ml2.Source = source1;
+            ml2.Receiver = receiver2;
+
+            // Media line #2 is connected.
+            Assert.AreEqual(source1.MediaLines[1], ml2);
+            Assert.AreEqual(receiver2.MediaLine, ml2);
+
+            // Disable the peer.
+            pc.enabled = false;
+
+            // Add a media line while disabled.
+            VideoReceiver receiver3 = go.AddComponent<VideoReceiver>();
+            MediaLine ml3 = pc.AddMediaLine(MediaKind.Video);
+            ml3.Source = source1;
+            ml3.Receiver = receiver3;
+
+            // Media line #3 is connected.
+            Assert.AreEqual(source1.MediaLines[2], ml3);
```

Comment:
And with active but not enabled it does add to `MediaLines`? This is so confusing.

Reply:
_Awake_ but not enabled. This is kind of arbitrary, we could do it the other way (only call the callbacks if the peer is enabled rather than if it's awake). I thought it might be better to limit the cases where `MediaLines` is not populated to when the peer is not awake which is kind of a niche situation, so this way looked a bit better. No strong feelings about it.

Reply:
See other comment, leave as is.